apps/frontend/src/pages/analytics/analytics.tsx

```diff
@@ -65,13 +81,16 @@ const Analytics = (): JSX.Element => {
 	);
 
 	useEffect(() => {
-		handleLoadLogs(dateRangeValue, projectValue);
-	}, [dateRangeValue, projectValue, handleLoadLogs]);
+		handleLoadLogs(dateRangeValue, projectValueString);
+	}, [dateRangeValue, projectValue, handleLoadLogs, projectValueString]);
 
 	const handleFormSubmit = useCallback(
 		(event_?: React.BaseSyntheticEvent): void => {
 			void handleSubmit((formData) => {
-				handleLoadLogs(formData.dateRange, formData.project);
+				handleLoadLogs(
+					formData.dateRange,
+					formData.project === null ? "" : formData.project.toString(),
+				);
```

Comment:
let's pass it this way, we need to convert project id to string only in place where we pass it to query parameters (action). Also we don't need this check for null. We need to include this parameter conditionally inside the action 
```suggestion
			void handleSubmit(({ dateRange, projectId }) => {
				handleLoadLogs({
				  dateRange,
				  projectId
				});
```

Reply:
There are some complications with validation schema if I change the types so action receives number instead of string. Hope this is ok
```
const handleLoadLogs = useCallback(
		([startDate, endDate]: [Date, Date], projectId?: null | number) => {
			const formattedStartDate = startDate.toISOString();
			const formattedEndDate = endDate.toISOString();

			void dispatch(
				activityLogActions.loadAll({
					endDate: formattedEndDate,
					projectId: projectId?.toString() ?? undefined,
					startDate: formattedStartDate,
				}),
			);
		},
		[dispatch],
	);
```